### PR TITLE
[ArtworkCarouselHeader] Fix crash due to missing artist details.

### PR DESCRIPTION
**Pull request by alloy**
https://artsyproduct.atlassian.net/browse/IN-62 #resolve

#skip_new_tests

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Scenes/Home/Components/ForYou/Components/__tests__/ArtworkCarouselHeader-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/Scenes/Home/Components/ForYou/Components/ArtworkCarouselHeader.tsx [hunk 1]
```diff
@@ -177,15 +182,23 @@ interface RelayProps {
   rail: {
     title: string | null
     key: string | null
-    context: any
-  }
-}
-
-// FIXME: What is this workaround? Can we stop using it?
```
Comment (sarahscott) [inline src/lib/Scenes/Home/Components/ForYou/Components/ArtworkCarouselHeader.tsx]: 🎉  for fixing a FIXME

File: src/lib/Scenes/Home/Components/ForYou/Components/ArtworkCarouselHeader.tsx [hunk 2]
```diff
@@ -108,8 +108,6 @@ class ArtworkCarouselHeader extends Component<Props & RelayPropsWorkaround, Stat
     ARTemporaryAPIModule.setFollowArtistStatus(!this.state.following, context.artist.id, (error, following) => {
       if (error) {
         console.error("ArtworkCarouselHeader.tsx", error)
-      } else {
```
Comment (sarahscott) [inline src/lib/Scenes/Home/Components/ForYou/Components/ArtworkCarouselHeader.tsx]: Just so I understand what the bug was - this code was executing even if the artist didn't exist, and the above check for `HomePageModuleContextRelatedArtist` ensures the rail definitely has artist details?

Comment (alloy) [reply src/lib/Scenes/Home/Components/ForYou/Components/ArtworkCarouselHeader.tsx]: This method would execute, yeah, but the crash would occur on [line 102](https://github.com/artsy/emission/pull/966/files#diff-b3db040e2aa0a16c5d8a4f98d02c0ae5R102) where `context.artist` was `undefined` when this module represented a ‘followed artist’, as the data for that was not being queried for. That query is now added to the fragment.

**Pull request merged**